### PR TITLE
Print kubectl debug messages received when starting a container

### LINKED ISSUE #108135: ephemeral containers: show more information on failures to start containers
### What happened?

```
$ kubectl debug -it shell-6874dcdd7f-rv5vd --image bogus
Defaulting debug container name to debugger-r2gq7.
... hang forever ...
```

`kubectl describe` shows there are, as expected, failures to pull the image. It would be great if this - and possibly other issues of starting the container - were exposed through the `kubectl debug` call to make things more obvious to the users

### What did you expect to happen?

More info is shown to user

### How can we reproduce it (as minimally and precisely as possible)?

See above

### Anything else we need to know?

_No response_

### Kubernetes version

<details>

```console
$ kubectl version
Client Version: version.Info{Major:"1", Minor:"23", GitVersion:"v1.23.3", GitCommit:"816c97ab8cff8a1c72eccca1026f7820e93e0d25", GitTreeState:"clean", BuildDate:"2022-01-25T21:25:17Z", GoVersion:"go1.17.6", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"23", GitVersion:"v1.23.2-gke.300", GitCommit:"516d535fbe1cce83a6ac4dfbaa749b3c3ed499ac", GitTreeState:"clean", BuildDate:"2022-01-24T09:29:03Z", GoVersion:"go1.17.5b7", Compiler:"gc", Platform:"linux/amd64"}

```

</details>

### Cloud provider

<details>
GKE
</details>

### OS version

<details>

```console
# On Linux:
$ cat /etc/os-release
# paste output here
$ uname -a
# paste output here

# On Windows:
C:\> wmic os get Caption, Version, BuildNumber, OSArchitecture
# paste output here
```

</details>

### Install tools

<details>

</details>

### Container runtime (CRI) and and version (if applicable)

<details>

</details>

### Related plugins (CNI, CSI, ...) and versions (if applicable)

<details>

</details>

### PR DESCRIPTION
**Pull request by verb**
#### What type of PR is this?

/kind bug

#### What this PR does / why we need it:

This provides feedback to the user, for example that the server is unable to pull the debug container image, for example:

```
$ kubectl debug -it --image=bogus shared      
Defaulting debug container name to debugger-rfdmp.
Container debugger-rfdmp: rpc error: code = Unknown desc = failed to pull and unpack image "docker.io/library/bogus:latest": failed to resolve reference "docker.io/library/bogus:latest": pull access denied, repository does not exist or may require authorization: server message: insufficient_scope: authorization failed
Container debugger-rfdmp: Back-off pulling image "bogus"
```

#### Which issue(s) this PR fixes:

Fixes #108135

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: staging/src/k8s.io/kubectl/pkg/cmd/debug/debug.go [hunk 1]
```diff
@@ -740,6 +741,9 @@ func waitForContainer(ctx context.Context, podClient corev1client.PodsGetter, ns
 			if s.State.Running != nil || s.State.Terminated != nil {
 				return true, nil
 			}
+			if s.State.Waiting != nil && s.State.Waiting.Message != "" {
+				fmt.Printf("Container %s: %s\n", containerName, s.State.Waiting.Message)
```
Comment (VirrageS) [inline staging/src/k8s.io/kubectl/pkg/cmd/debug/debug.go:745]: For consistency I think this should be:
```suggestion
				fmt.Fprintf(o.Out, "Container %s: %s\n", containerName, s.State.Waiting.Message)
```
but for that you need to pass additional argument to `waitForContainer`.

Comment (sftim) [reply staging/src/k8s.io/kubectl/pkg/cmd/debug/debug.go:745]: Are these printed to stderr or stdout?

What I'd prefer: stderr
What I'd like as an extra: detect if the console supports color, and if so color these messages so they stand apart from the container's own stderr.

Comment (verb) [reply staging/src/k8s.io/kubectl/pkg/cmd/debug/debug.go:745]: It's stdout currently, but I agree that stderr makes more sense. I'll plumb the streams through.

Re: colors, I think that's a cool idea but not one I'd like to take on right now unless there's a library in kubectl that already supports it. I'd be happy to make it clearer textually that the messages is being printed by kubectl, though.

Comment (verb) [reply staging/src/k8s.io/kubectl/pkg/cmd/debug/debug.go:745]: (and thanks for these suggestions @sftim & @VirrageS)

Comment (verb) [reply staging/src/k8s.io/kubectl/pkg/cmd/debug/debug.go]: fixed, please take another look.

Comment (VirrageS) [reply staging/src/k8s.io/kubectl/pkg/cmd/debug/debug.go]: LGTM

Comment (eddiezane) [reply staging/src/k8s.io/kubectl/pkg/cmd/debug/debug.go]: Warnings will be overhauled in https://github.com/kubernetes/kubernetes/pull/108210

File: staging/src/k8s.io/kubectl/pkg/cmd/debug/debug.go [hunk 2]
```diff
@@ -740,6 +741,9 @@ func waitForContainer(ctx context.Context, podClient corev1client.PodsGetter, ns
 			if s.State.Running != nil || s.State.Terminated != nil {
 				return true, nil
 			}
+			if !o.Quiet && s.State.Waiting != nil && s.State.Waiting.Message != "" {
+				fmt.Fprintf(o.ErrOut, "Container %s: %s\n", containerName, s.State.Waiting.Message)
```
Comment (eddiezane) [inline staging/src/k8s.io/kubectl/pkg/cmd/debug/debug.go:745]: To make this easier to find later for https://github.com/kubernetes/kubernetes/pull/108210
```suggestion
				fmt.Fprintf(o.ErrOut, "Warning: container %s: %s\n", containerName, s.State.Waiting.Message)
```

**Review by eddiezane — CHANGES_REQUESTED** [review]

**Review by eddiezane — APPROVED** [review]
Thanks!

/lgtm
/approve
/label tide/merge-method-squash

**Pull request merged**